python/pyarrow/tests/test_flight.py

```diff
@@ -1810,3 +1813,62 @@ def test_generic_options():
                                 generic_options=options)
         with pytest.raises(pa.ArrowInvalid):
             client.do_get(flight.Ticket(b'ints'))
+
+
+class CancelFlightServer(FlightServerBase):
+    """A server for testing StopToken."""
+
+    def do_get(self, context, ticket):
+        schema = pa.schema([])
+        rb = pa.RecordBatch.from_arrays([], schema=schema)
+        return flight.GeneratorStream(schema, itertools.repeat(rb))
+
+    def do_exchange(self, context, descriptor, reader, writer):
+        schema = pa.schema([])
+        rb = pa.RecordBatch.from_arrays([], schema=schema)
+        writer.begin(schema)
+        while not context.is_cancelled():
+            # TODO: writing schema.empty_table() here hangs/fails
+            writer.write_batch(rb)
```

Comment:
Would this raise an exception if the context is cancelled, or just succeed immediately?

Reply:
It should raise an exception, but either way, the client is gone at this point so there's not much the server can do.

Reply:
Ok, I was just wondering if the server could end up spending more time than necessary here :-)